socorro/external/boto/crashstorage.py

```diff
@@ -76,102 +88,87 @@ def do_save_raw_crash(boto_connection, raw_crash, dumps, crash_id):
         for dump_name, dump in iteritems(dumps):
             if dump_name in (None, "", "upload_file_minidump"):
                 dump_name = "dump"
-            boto_connection.submit(crash_id, dump_name, dump)
+            conn.submit(crash_id, dump_name, dump)
 
     def save_raw_crash(self, raw_crash, dumps, crash_id):
-        retry(
-            self.connection_source,
-            self.quit_check,
-            self.do_save_raw_crash,
-            raw_crash=raw_crash,
-            dumps=dumps,
-            crash_id=crash_id,
-        )
+        """Save raw_crash, dump_names, and dumps."""
+        retryable_save = self.retryable(self.do_save_raw_crash)
+        with self.connection_source() as conn:
+            retryable_save(conn, raw_crash=raw_crash, dumps=dumps, crash_id=crash_id)
```

Comment:
This is functionally similar to the old code except it doesn't use the `quit_check_callback` stuff. Because the wait time generator is `1, 1, 1, 1, 1`, it waits at most 5 seconds. Given that, I decided it's ok to drop the quit check callback stuff for now.

Reply:
I agree, losing interactive abilities for 5 seconds is probably not a big deal. I think it would be possible to port ``quit_check_callback`` by implementing a  ``sleep_function`` for ``retry``.